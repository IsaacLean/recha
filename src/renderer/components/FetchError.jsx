// @flow
import React from 'react'

const FetchError = ({ onRetry, message }: { onRetry: () => void, message: string}) => (
  <div>
    <p>Could not fetch todos. {message}</p>
    <button onClick={onRetry}>Retry</button>
  </div>
)

export default FetchError
