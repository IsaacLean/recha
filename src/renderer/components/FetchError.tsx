import React from 'react'

const FetchError = ({ onRetry, message }: { onRetry: () => void; message: string }): JSX.Element => (
  <div>
    <p>Could not fetch todos. {message}</p>
    <button onClick={onRetry}>Retry</button>
  </div>
)

export default FetchError
