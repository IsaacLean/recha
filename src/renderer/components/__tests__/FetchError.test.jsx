import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import FetchError from '../FetchError'

Enzyme.configure({ adapter: new Adapter() })

describe('FetchError', () => {
  it('renders properly', () => {
    const component = shallow(<FetchError />)
    expect(component).toMatchSnapshot()
  })
})
