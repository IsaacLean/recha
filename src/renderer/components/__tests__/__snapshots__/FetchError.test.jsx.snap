// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchError renders properly 1`] = `
<div>
  <p>
    Could not fetch todos. 
  </p>
  <button>
    Retry
  </button>
</div>
`;
